File: ABC_Chapter10_Exercise2/app/src/main/java/com/example/cyu/abc_chapter10_exercise1/MainActivity.java
package com.example.cyu.abc_chapter10_exercise1;

import android.graphics.drawable.AnimationDrawable;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;

public class MainActivity extends AppCompatActivity {

    AnimationDrawable lightsAnimation;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ImageView imgFrame =  findViewById(R.id.imgGolf);
        imgFrame.setBackgroundResource(R.drawable.animation);
        lightsAnimation = (AnimationDrawable) imgFrame.getBackground();

        Button button1 = findViewById(R.id.btnStart);
        button1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                lightsAnimation.start();
            }
        });
    }
}

File: ABC_Chapter10_Exercise2/app/src/main/java/com/example/cyu/abc_chapter10_exercise1/MainActivity.kt
package com.example.cyu.abc_chapter10_exercise1

import android.graphics.drawable.AnimationDrawable
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.ImageView

class MainActivity : AppCompatActivity() {

    private lateinit var golfAnimation: AnimationDrawable

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val imgFrame = findViewById<ImageView>(R.id.imgGolf)
        imgFrame.setBackgroundResource(R.drawable.animation)

        golfAnimation = imgFrame.background as AnimationDrawable

        val button1 = findViewById<Button>(R.id.btnStart)
        button1.setOnClickListener { golfAnimation.start() }
    }
}
